src/tests/compiler/ast/module/sourceFileTests.ts

```diff
@@ -1300,6 +1300,20 @@ function myFunction(param: MyClass) {
                 { path: "/MyInterface.ts", text: "export interface MyInterface {}" }
             ], expectedText);
         });
+
+        it("should not forget nodes", () => {
+            const startText = `import { MyClass } from "./MyClass";\n\nconst t = new MyClass();\nconst u: MyInterface = {};\nconst v = new MyClass2();\nconst w = new MyClass3()`;
+            const { sourceFile, project } = getInfoFromText(startText);
+            project.createSourceFile("/MyClass.ts", "export class MyClass {} export class MyClass2 {} export class MyClass3 {}");
+            project.createSourceFile("/MyInterface.ts", "export interface MyInterface {}");
+            const w = sourceFile.getVariableDeclaration("w")!.getFirstDescendantByKindOrThrow(SyntaxKind.Identifier);
+            const i = sourceFile.getImportDeclarations()[0];
+            expect(i.getText()).to.equals("import { MyClass } from \"./MyClass\";");
+            sourceFile.fixMissingImports();
+            expect(() => w.getText()).not.to.throw();
```

Comment:
Check `w.wasForgotten()`. It's a bit more explicit. Some old places might be checking if it throws, but that was before I added `wasForgotten()`.